src/components/storage/InstallationScenario.jsx

```diff
@@ -92,7 +99,56 @@ const checkMountPointMapping = ({ hasFilesystems, duplicateDeviceNames }) => {
     return availability;
 };
 
-const scenarios = [{
+export const checkConfiguredStorage = ({ mountPointConstraints, partitioning, newMountPoints, scenarioPartitioningMapping }) => {
+    const availability = new AvailabilityState();
+
+    const currentPartitioningMatches = partitioning !== undefined && scenarioPartitioningMapping["use-configured-storage"] === partitioning;
+    availability.hidden = partitioning === undefined || !currentPartitioningMatches;
+
+    availability.available = (
+        newMountPoints === undefined ||
+        (
+            mountPointConstraints
+                    ?.filter(m => m.required.v)
+                    .every(m => {
+                        const allDirs = [];
+                        const getNestedDirs = (object) => {
+                            if (!object) {
+                                return;
+                            }
+                            const { dir, content, subvolumes } = object;
+
+                            if (dir) {
+                                allDirs.push(dir);
+                            }
+                            if (content) {
+                                getNestedDirs(content);
+                            }
+                            if (subvolumes) {
+                                Object.keys(subvolumes).forEach(sv => getNestedDirs(subvolumes[sv]));
+                            }
+                        };
+
+                        if (m["mount-point"].v) {
+                            Object.keys(newMountPoints).forEach(key => getNestedDirs(newMountPoints[key]));
+
+                            return allDirs.includes(m["mount-point"].v);
+                        }
+
+                        // FIXME: cockpit does not return the biosboot partitions in the cockpit_mount_points
```

Comment:
Why would it? A biosboot partition has no mount point. Doesn't blivet find them based on their partition type?

Reply:
Yes I will remove the comment.